File: scripts/generate-placeholder.js
const { createCanvas } = require('canvas');
const fs = require('fs');
const path = require('path');

function generatePlaceholder() {
    const width = 4096;
    const height = 2048;
    const canvas = createCanvas(width, height);
    const ctx = canvas.getContext('2d');

    // Sky gradient (top half)
    const skyGradient = ctx.createLinearGradient(0, 0, 0, height / 2);
    skyGradient.addColorStop(0, '#1e3a8a');
    skyGradient.addColorStop(0.5, '#3b82f6');
    skyGradient.addColorStop(1, '#60a5fa');
    ctx.fillStyle = skyGradient;
    ctx.fillRect(0, 0, width, height / 2);

    // Ground/water gradient (bottom half)
    const groundGradient = ctx.createLinearGradient(0, height / 2, 0, height);
    groundGradient.addColorStop(0, '#059669');
    groundGradient.addColorStop(0.3, '#10b981');
    groundGradient.addColorStop(1, '#065f46');
    ctx.fillStyle = groundGradient;
    ctx.fillRect(0, height / 2, width, height / 2);

    // Draw buildings
    const buildings = [
        { x: 1500, y: 700, w: 100, h: 350 },
        { x: 1650, y: 600, w: 120, h: 450 },
        { x: 1820, y: 650, w: 90, h: 400 },
        { x: 1950, y: 720, w: 110, h: 330 },
        { x: 2100, y: 680, w: 100, h: 370 },
        { x: 2250, y: 550, w: 140, h: 500 },
        { x: 2440, y: 700, w: 95, h: 350 },
    ];

    buildings.forEach((building, idx) => {
        // Building body
        const colors = ['#1f2937', '#374151', '#4b5563'];
        ctx.fillStyle = colors[idx % colors.length];
        ctx.fillRect(building.x, building.y, building.w, building.h);

        // Windows
        ctx.fillStyle = '#fbbf24';
        const rows = Math.floor(building.h / 30);
        const cols = Math.floor(building.w / 20);
        for (let i = 0; i < rows; i++) {
            for (let j = 0; j < cols; j++) {
                if (Math.random() > 0.3) {
                    ctx.fillRect(
                        building.x + j * 20 + 5,
                        building.y + i * 30 + 10,
                        10,
                        15
                    );
                }
            }
        }
    });

    // Add sun
    ctx.fillStyle = '#fbbf24';
    ctx.beginPath();
    ctx.arc(3500, 300, 80, 0, Math.PI * 2);
    ctx.fill();

    // Add text
    ctx.fillStyle = 'white';
    ctx.font = 'bold 80px Arial';
    ctx.textAlign = 'center';
    ctx.fillText('AUSTIN SKYLINE', width / 2, 400);
    ctx.font = '50px Arial';
    ctx.fillText('Placeholder - Replace with Real 360° Panorama', width / 2, 500);

    // Save to file
    const outputPath = path.join(__dirname, '..', 'public', 'austin-panorama.jpg');
    const buffer = canvas.toBuffer('image/jpeg', { quality: 0.85 });
    fs.writeFileSync(outputPath, buffer);

    console.log('✅ Placeholder panorama generated!');
    console.log(`   Location: ${outputPath}`);
    console.log(`   Size: ${width}x${height} pixels`);
    console.log('\n⚠️  Replace this with a real 360° panorama for production!');
}

// Run if this script is executed directly
if (require.main === module) {
    try {
        generatePlaceholder();
    } catch (error) {
        console.error('Error generating placeholder:', error.message);
        console.log('\nNote: This script requires the "canvas" package.');
        console.log('Install it with: npm install canvas');
        process.exit(1);
    }
}

module.exports = { generatePlaceholder };

File: scripts/generate_placeholder.py
#!/usr/bin/env python3
"""
Generate a placeholder panorama image for testing the Austin panorama viewer.
Requires: pip install pillow
"""

from PIL import Image, ImageDraw, ImageFont
import random

def generate_placeholder_panorama(width=4096, height=2048):
    """Generate a placeholder equirectangular panorama"""

    # Create image
    img = Image.new('RGB', (width, height))
    draw = ImageDraw.Draw(img)

    # Sky gradient (top half)
    for y in range(height // 2):
        ratio = y / (height // 2)
        r = int(30 + ratio * 66)
        g = int(58 + ratio * 72)
        b = int(138 + ratio * 112)
        draw.rectangle([(0, y), (width, y + 1)], fill=(r, g, b))

    # Ground/water gradient (bottom half)
    for y in range(height // 2, height):
        ratio = (y - height // 2) / (height // 2)
        if ratio < 0.3:  # Water
            r = int(5 + ratio * 11)
            g = int(150 + ratio * 35)
            b = int(105 - ratio * 24)
        else:  # Grass
            r = int(6 + ratio * 0)
            g = int(95 + ratio * 90)
            b = int(46 + ratio * 24)
        draw.rectangle([(0, y), (width, y + 1)], fill=(r, g, b))

    # Draw buildings for Austin skyline
    buildings = [
        {'x': int(width * 0.35), 'y': int(height * 0.35), 'w': 80, 'h': 280},
        {'x': int(width * 0.38), 'y': int(height * 0.30), 'w': 100, 'h': 360},
        {'x': int(width * 0.42), 'y': int(height * 0.32), 'w': 75, 'h': 320},
        {'x': int(width * 0.45), 'y': int(height * 0.36), 'w': 90, 'h': 270},
        {'x': int(width * 0.50), 'y': int(height * 0.34), 'w': 85, 'h': 300},
        {'x': int(width * 0.53), 'y': int(height * 0.28), 'w': 110, 'h': 400},  # Tall building (Frost Bank)
        {'x': int(width * 0.57), 'y': int(height * 0.36), 'w': 80, 'h': 280},
    ]

    for building in buildings:
        # Building body
        color = (random.randint(31, 75), random.randint(41, 83), random.randint(87, 111))
        draw.rectangle(
            [(building['x'], building['y']),
             (building['x'] + building['w'], building['y'] + building['h'])],
            fill=color
        )

        # Windows
        window_color = (251, 191, 36) if random.random() > 0.5 else (200, 200, 100)
        window_rows = building['h'] // 25
        window_cols = building['w'] // 15

        for i in range(window_rows):
            for j in range(window_cols):
                if random.random() > 0.2:  # 80% chance of lit window
                    wx = building['x'] + j * 15 + 4
                    wy = building['y'] + i * 25 + 8
                    draw.rectangle([(wx, wy), (wx + 8, wy + 12)], fill=window_color)

    # Add sun
    sun_x = int(width * 0.85)
    sun_y = int(height * 0.15)
    draw.ellipse([(sun_x - 60, sun_y - 60), (sun_x + 60, sun_y + 60)],
                 fill=(251, 191, 36))

    # Add text overlay
    try:
        font_large = ImageFont.truetype("/System/Library/Fonts/Helvetica.ttc", 100)
        font_small = ImageFont.truetype("/System/Library/Fonts/Helvetica.ttc", 60)
    except:
        font_large = ImageFont.load_default()
        font_small = ImageFont.load_default()

    text1 = "AUSTIN SKYLINE"
    text2 = "Placeholder - Replace with Real Panorama"

    # Get text bounding boxes
    bbox1 = draw.textbbox((0, 0), text1, font=font_large)
    bbox2 = draw.textbbox((0, 0), text2, font=font_small)

    text1_width = bbox1[2] - bbox1[0]
    text2_width = bbox2[2] - bbox2[0]

    # Draw text with shadow
    shadow_offset = 4
    text1_x = (width - text1_width) // 2
    text1_y = int(height * 0.20)

    draw.text((text1_x + shadow_offset, text1_y + shadow_offset), text1,
              fill=(0, 0, 0), font=font_large)
    draw.text((text1_x, text1_y), text1, fill=(255, 255, 255), font=font_large)

    text2_x = (width - text2_width) // 2
    text2_y = int(height * 0.25)

    draw.text((text2_x + shadow_offset, text2_y + shadow_offset), text2,
              fill=(0, 0, 0), font=font_small)
    draw.text((text2_x, text2_y), text2, fill=(255, 255, 255), font=font_small)

    return img

if __name__ == '__main__':
    print("Generating placeholder panorama image...")
    img = generate_placeholder_panorama()

    output_path = '../public/austin-panorama.jpg'
    img.save(output_path, 'JPEG', quality=85, optimize=True)

    print(f"✅ Placeholder panorama saved to: {output_path}")
    print(f"   Size: {img.size[0]}x{img.size[1]} pixels")
    print("\nReplace this with a real 360° panorama photo for production!")
